Dockerfile

```diff
@@ -16,22 +16,22 @@ RUN if [ "$SYSTEM" = "gpu" ]; then \
     wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb && \
     dpkg -i cuda-keyring_1.1-1_all.deb && \
     apt-get update && apt-get install -y --no-install-recommends \
-    cuda-command-line-tools-11-8 \
-    cuda-cudart-dev-11-8 \
-    cuda-nvcc-11-8 \
-    cuda-cupti-11-8 \
-    cuda-nvprune-11-8 \
-    cuda-libraries-11-8 \
-    cuda-nvrtc-11-8 \
-    libcufft-11-8 \
-    libcurand-11-8 \
-    libcusolver-11-8 \
-    libcusparse-11-8 \
-    libcublas-11-8 \
+    cuda-command-line-tools-12-3 \
+    cuda-cudart-dev-12-3 \
+    cuda-nvcc-12-3 \
+    cuda-cupti-12-3 \
+    cuda-nvprune-12-3 \
+    cuda-libraries-12-3 \
+    cuda-nvrtc-12-3 \
+    libcufft-12-3 \
+    libcurand-12-3 \
+    libcusolver-12-3 \
+    libcusparse-12-3 \
+    libcublas-12-3 \
     # - CuDNN: https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html#ubuntu-network-installation
-    libcudnn8=8.6.0.163-1+cuda11.8 \
-    libnvinfer-plugin8=8.6.1.6-1+cuda11.8 \
-    libnvinfer8=8.6.1.6-1+cuda11.8; \
+    libcudnn8=8.9.6.50-1+cuda12.2 \
```

Comment:
I think this should be `libcudnn9` !? 
I remember this was the main issue i had also with OnnxTR running with cuda >= 12
ref.: https://hub.docker.com/layers/nvidia/cuda/12.3.2-cudnn9-runtime-ubuntu20.04/images/sha256-5bb3eeae860aa89abd6f133c68eb15eea08d784a8655aa79d55aa450b2961d63?context=explore

Would it be maybe easier to use a already working base image ?

I did something similar for our pipelines

```
ARG IMAGE=nvidia/cuda:12.6.2-base-ubuntu22.04

FROM ${IMAGE}
ARG SYSTEM=gpu
ARG PORT=80
ARG GITHUB_TOKEN

WORKDIR /app
ENV PATH="/root/miniconda3/bin:${PATH}"
ARG PATH="/root/miniconda3/bin:${PATH}"
ENV STORAGE_PATH ${STORAGE_PATH}
ENV PORT ${PORT}
ENV GITHUB_TOKEN ${GITHUB_TOKEN}

# if STORAGE_PATH is not empty set it as VOLUME otherwise set "/simplification-storage"
VOLUME ${STORAGE_PATH:-/simplification-storage}


COPY pyproject.toml /app/pyproject.toml
COPY app /app/app

RUN apt-get -y update
RUN apt-get -y install git
RUN apt-get install -y wget && rm -rf /var/lib/apt/lists/*

RUN wget \
    https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
    && mkdir /root/.conda \
    && bash Miniconda3-latest-Linux-x86_64.sh -b \
    && rm -f Miniconda3-latest-Linux-x86_64.sh

RUN if [ "$SYSTEM" = "gpu" ]; then \
    conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia ; \
    else \
    conda install pytorch torchvision torchaudio cpuonly -c pytorch ; \
    fi

RUN pip install . --extra-index-url=https://XYZ

# Echo some information
RUN echo ${PORT} \
    && echo ${SYSTEM} \
    && echo ${STORAGE_PATH}

EXPOSE ${PORT}

CMD uvicorn app.main:app --host 0.0.0.0 --port ${PORT}
```

In my case default is used for the kubernetes cluster and the following for the GH CI testing stuff

```
docker compose build --build-arg "IMAGE=python:3.10-slim" --build-arg "SYSTEM=cpu" --build-arg "PORT=8000"
```

Reply:
@felixdittrich92 You install `pytorch-cuda=11.8` in Dockerfile, I think it'll mess up the nvidia driver in the Docker container

Reply:
Yeah, indeed, it would be easier but I think these images are bloated. Do you know where I can find its Dockerfile, I can't find it.
With the updated version, PyTorch works but not TensorFlow :facepalm:
Moreover, both frameworks provides nvidia drivers now, so I think we need to revamp a bit the Dockerfile and the `pyproject.toml`.
I'll continue this PR, I'll not have too much time today to work on it

Reply:
@odulcy-mindee Not sure i think it's this: https://gitlab.com/nvidia/container-images/cuda/-/blob/master/templates/ubuntu/base-dockerfile.j2

Reply:
Ähhmm.. i think with tensorflow it changed a bit 

pip3 install tensorflow  does not anymore provide cuda support by default it should be `tensorflow[and-cuda]` now, if the problem is that it does not take the GPU

Reply:
Yeah no stress --slow progress is better as no progress 🤣

Reply:
And this one looks like the latest: https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/12.6.1/ubuntu2204/base/Dockerfile?ref_type=heads

Reply:
Ah, thank you, the latter seems simpler. I wanted to have a lighter docker image for CPU. I guess simpler version would be to use `nvidia/cuda:12.4.0-base-ubuntu22.04`.

Reply:
```
- name: Build Docker image
        id: build
        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
        with:
          context: .
          build-args: |
            IMAGE=${{ matrix.base_image }}
            FRAMEWORK=${{ matrix.framework }}
            PYTHON_VERSION=${{ matrix.python }}
            SYSTEM=${{ matrix.system }}
            DOCTR_REPO=${{ github.repository }}
            DOCTR_VERSION=${{ github.sha }}
```

Reply:
Ah i see this would conflict with the python version arg

Reply:
And for the tensorflow issue i think the only option would be to add an additional specifier:

```
tf = [
    # cf. https://github.com/mindee/doctr/pull/1461
    "tensorflow>=2.15.0,<3.0.0",
    "tf-keras>=2.15.0,<3.0.0",  # Keep keras 2 compatibility
    "tf2onnx>=1.16.0,<2.0.0",  # cf. https://github.com/onnx/tensorflow-onnx/releases/tag/v1.16.0
]
tf-gpu = [
    # cf. https://github.com/mindee/doctr/pull/1461
    "tensorflow[and-cuda]>=2.15.0,<3.0.0",
    "tf-keras>=2.15.0,<3.0.0",  # Keep keras 2 compatibility
    "tf2onnx>=1.16.0,<2.0.0",  # cf. https://github.com/onnx/tensorflow-onnx/releases/tag/v1.16.0
]
```

Which would also require updates in the readme / docs / (CI / API ?) 😅

Reply:
> the only option would be to add an additional specifier
Yeah, I guess so

Reply:
Pseudo: 

Something like the following could work:

```
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        base_image: ["nvidia/cuda:12.4.0-base-ubuntu22.04", "python:py-version-slim"]
        python: ["3.10.13", "3.11.8", "3.12.7"]
    
    steps:
      - name: Set IMAGE
        id: image
        run: |
          if [[ "${{ matrix.base_image }}" == *"py-version"* ]]; then
            echo "IMAGE=${{ matrix.base_image/py-version/${{ matrix.python }} }}" >> $GITHUB_ENV
          else
            echo "IMAGE=${{ matrix.base_image }}" >> $GITHUB_ENV

      - name: Build Docker image
        id: build
        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
        with:
          context: .
          build-args: |
            IMAGE=${{ env.IMAGE }}
            FRAMEWORK=${{ matrix.framework }}
            PYTHON_VERSION=${{ matrix.python }}   # Only required for the nvidia base image
            SYSTEM=${{ matrix.system }}  # I think can be dropped in this case
            DOCTR_REPO=${{ github.repository }}
            DOCTR_VERSION=${{ github.sha }}
 ```